src/NHibernate/Cfg/Configuration.cs

```diff
@@ -941,11 +945,12 @@ public string[] GenerateSchemaCreationScript(Dialect.Dialect dialect)
 
 			var script = new List<string>();
 
+			var staticDialectMapping = new StaticDialectMappingWrapper(mapping, dialect);
```

Comment:
You would potentially need to do same as in `BuildSessionFactory` if you want to leave `mapping` field in place: replace `mapping` field with wrapped instance and back when operation finished, otherwise something can access the unwrapped `mapping` field and then have different dialect than was configured. 

Even if there are no places that would be affected now - this could be in the future. 

This is the reason I've decided to remove `mapping` field altogether.

Reply:
As a fix of previous versions, it is good enough for me.